neutrino.go

```diff
@@ -66,6 +67,10 @@ var (
 	// DisableDNSSeed disables getting initial addresses for Bitcoin nodes
 	// from DNS.
 	DisableDNSSeed = false
+
+	// DefaultFilterCacheSize is the number of filters neutrino will keep
```

Comment:
Perhaps we should instead target a size in memory usage? 

This would allow more precise tuning of the cache, as the size of each filter varies based on the composition of the block. In order to do this, we may need to instead create a new interface and use that as the input to the cache interface for each of the methods. The cache would store this type `T`. `T` has a single method, just to compute the size of the element serialized in bytes. This will allow the cache to prune based on size rather than number of elements.

Reply:
yes, that makes a lot of sense.

so essentially the `size` in LRUCache becomes number of bytes instead of elements.

when a new element is inserted, we check that it's size isn't bigger than the size of the cache, otherwise evict as many filters as necessary to be able to put this one in (of course there is the issue that we could evict many smaller filters instead of a single big filter, but different tuning could be used later I guess).

one thing I don't really understand is (I haven't written a lot of go code...) why do we need a separate interface with the `ComputeSize` method ? wouldn't len(filter.filterData) work ? or filter could have a `Size` method ?

Reply:
A `ComputeSize` interface is preferred since the cache can then be used for arbitrary types :) I agree that `Size` is a better name.